cnd/src/swap_protocols/rfc003/ethereum/htlc_events.rs

```diff
@@ -41,14 +42,26 @@ impl
 {
     async fn htlc_funded(
         &self,
-        _htlc_params: &HtlcParams<Ethereum, asset::Ether, identity::Ethereum>,
+        htlc_params: &HtlcParams<Ethereum, asset::Ether, identity::Ethereum>,
         deploy_transaction: &Deployed<htlc_location::Ethereum, transaction::Ethereum>,
         _start_of_swap: NaiveDateTime,
     ) -> anyhow::Result<Funded<asset::Ether, transaction::Ethereum>> {
-        Ok(Funded {
-            asset: Ether::from_wei(deploy_transaction.transaction.value),
-            transaction: deploy_transaction.transaction.clone(),
-        })
+        let expected_asset = &htlc_params.asset;
+
+        let asset = Ether::from_wei(deploy_transaction.transaction.value);
+
+        let event = match expected_asset.cmp(&asset) {
```

Comment:
At first I liked this because we no longer have to specify the `Ord` bound but it is annoying that we have to do this for every asset now :(

Can you please record this as an issue so we don't forget to deal with this. I feel this should really be done in a more generic place.